core/src/main/java/com/ugamdemo/core/models/Impl/SystemuserImpl.java

```diff
@@ -0,0 +1,69 @@
+package com.ugamdemo.core.models.Impl;
+
+import com.day.cq.search.PredicateGroup;
+import com.day.cq.search.Query;
+import com.day.cq.search.QueryBuilder;
+import com.day.cq.search.result.Hit;
+import com.day.cq.search.result.SearchResult;
+import com.ugamdemo.core.models.Systemuser;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.models.annotations.DefaultInjectionStrategy;
+import org.apache.sling.models.annotations.Model;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.jcr.Session;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+//import java.util.logging.Logger;
+import org.slf4j.Logger;
+
+@Model(adaptables = SlingHttpServletRequest.class,
+        adapters = Systemuser.class,
+        defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL
+)
+
+
+public class SystemuserImpl implements Systemuser {
+
+
+    private static final Logger LOG = LoggerFactory.getLogger(Systemuser.class);
+    @Inject
+    ResourceResolver resolver;
+    @Inject
+    QueryBuilder queryBuilder;
+    @Override
+    public String getUsers() {
+        LOG.info("\n Printing LOGS from inside CLass");
+        String listofusers = "";
+        Map<String, String> map = new HashMap<>();
+        map.put("path", "/home/users");
+        map.put("property", "jcr:primaryType");
+        map.put("property.value", "rep:User");
+        map.put("p.hits", "selective");
+        map.put("p.properties", "rep:principalName");
+        map.put("type", "rep:User");
+        try {
+
```

Comment:
service user session ?

Reply:
yes, its using service user session.

Reply:
you are not using ResolverUtil to get the service user session, check that